### ISSUE TITLE
Not Restoring Checkpoints

### ISSUE THREAD

**Issue by Seiwert** [llSourcell/tensorflow_chatbot#11]
Hey all,

I was wondering if any of you have been encountering issues with restoring to previous checkpoints? After running the program in train mode for a few hours (step 4500), I attempted to test my results. When I did, I noticed that "Created model with fresh parameters" showed up in the terminal. So, from reading through the code, to my understanding, this means my previous checkpoint wasn't loaded.

Along with that, when I went to "train" mode again, it said it created a new model as well. I believe there is something wrong with the create_model function where lines 110-113 in execute.py are not retrieving and loading the model correctly. Looking through TensorFlow documentation, I believe this is because my checkpoints are being saved in three files. A .index, .meta, and .data-00000-of-00001 are being created instead of a .ckpt as described in the documentation. (See screen-shot)
![image](https://cloud.githubusercontent.com/assets/4229602/20998775/ef6139b4-bcd5-11e6-90f7-dfb9658c10f9.png)

Is anyone else having this problem or knows how to fix it? Do I need to use a different restore or save function? If anyone has more experience with TensorFlow and could help I'd appreciate it.

**TL;DR: Unable to restore checkpoints because three files are being created instead of one .ckpt file?**

**Comment by Seiwert** [comment]
So I figured out what the problem was. It was this line of code (line 111 in execute.py):
  if ckpt and tf.gfile.Exists(ckpt.model_checkpoint_path): 

the tf.gfile.Exists() was always failing. I guess it looks for the exact file path which will always fail because checkpoint doesn't contain file extensions. Removing that condition fixed my issue.

Also, to answer my other questions. The restore function just needs a relative path and doesn't require extensions. It loads the three files as part of restoring the checkpoint based on the passed in relative file path.

**Comment by JorgeCeja** [comment]
I'm still getting an error 

raise ValueError("Restore called with invalid save path %s" % save_path)
ValueError: Restore called with invalid save path working_dir/seq2seq.ckpt-27000

What else did you change to get it working?

Thanks!~

**Comment by Seiwert** [comment]
Hmm, that's interesting. I actually never received that error. 

However, if I had to take a guess, you may want to check to make sure that your ckpt-27000 has all three files. Depending on when you terminate your program while it is training, you may only have the .data-00000-of-00001, or .index and not the .meta file. Or, only 1 of the files and not the other, it really depends where you stop it at. 

If I were you, I'd check to make sure all three of those files are there. If they are not there, then you will need to modify the checkpoint file in the working_dir to point to the last checkpoint (e.g. 26700) and delete those other 27000 checkpoint files. 

Hope this helps!!
